docs: add docs on general Terraform usage (#1263)
edgelesssys/constellation: docs/docs/architecture/orchestration.md

* a configuration file
* an ID file
* a Base64-encoded master secret

> * Terraform artifacts such as `terraform.tfstate`, stored in the `constellation-terraform` subdirectory

Reviewer: ```suggestion
* [Terraform artifacts](../reference/terraform.md) stored in the `constellation-terraform` subdirectory
```
If we do it like this ...

* a Kubernetes `kubeconfig` file.

> Constellation uses Terraform for infrastructure management. No setup of Terraform is needed. The CLI automatically fetches a copy of Terraform when required. The policy on usage of Terraform by Constellation can be found [here](../reference/terraform.md).

Reviewer: ```suggestion
```
... we may be able to just drop this one.

* a configuration file
* an ID file
* a Base64-encoded master secret

> * [Terraform artifacts](../reference/terraform.md), stored in the `constellation-terraform` subdirectory

Reviewer 2: Should we also mention the iam subfolder?

The `kubeconfig` file provides the credentials and configuration for connecting and authenticating to the API server.
Once configured, orchestrate the Kubernetes cluster via `kubectl`.

> Make sure to keep the Terraform subdirectory such as `constellation-terraform` in the workspace directory to be able to manage your cluster later on.
> Without it, you won't be able to modify or terminate your cluster.

Reviewer: ```suggestion
```
I'm not sure if this hint is necessary at all after explaining what the workspace directory is. What do you think?
